apps/src/gamelab/GameLabSprite.js

```diff
@@ -567,6 +579,18 @@ var AABBops = function (p5Inst, type, target, callback, modifyPosition=true) {
   return result;
 };
 
+/**
+ * Projects a vector onto the line parallel to a second vector, giving a third
+ * vector which is the orthogonal projection of that vector onto the line.
+ * @see https://en.wikipedia.org/wiki/Vector_projection
+ * @param {p5.Vector} a - vector being projected
+ * @param {p5.Vector} b - vector defining the projection target line.
+ * @returns {p5.Vector} projection of a onto the line parallel to b.
+ */
+function projectAontoB(a, b) {
+  return p5.Vector.mult(b, p5.Vector.dot(a, b) / p5.Vector.dot(b, b));
```

Comment:
I didn't take too long agonizing over the most efficient calculation here, so feel free to pick this apart.  I went with this one, which Wikipedia claims is slightly more efficient to compute:

![image](https://cloud.githubusercontent.com/assets/1615761/19460004/6cbe8154-948c-11e6-9d8a-c1e8a429f6e0.png)